### PR TITLE
chore: Add cargo test job to CI

### PR DESCRIPTION
**Pull request by klochowicz**
We started adding some unit tests in the Rust library, so they should be run on
the CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -83,6 +83,26 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup rust toolchain
+        run: rustup show
+      - uses: Swatinem/rust-cache@v2.1.0
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: "stable"
+      - name: Install FFI bindings
+        run: cargo install flutter_rust_bridge_codegen
```
Comment (holzeis) [inline .github/workflows/ci.yml:99]: oh boy, install ffi bindings again 🙈 we need to tackle https://github.com/itchysats/10101/issues/79

**Review by holzeis — APPROVED** [review]

File: rust/src/wallet.rs [hunk 2]
```diff
@@ -119,9 +119,11 @@ mod tests {
...
     fn wallet_support_for_different_bitcoin_networks() {
         let temp = temp_dir();
-        wallet::init_wallet(Network::Mainnet, &temp).expect("wallet to be initialized");
-        wallet::init_wallet(Network::Testnet, &temp).expect("wallet to be initialized");
-        wallet::init_wallet(Network::Regtest, &temp)
+        wallet::init_wallet(Network::Mainnet, &temp.join("mainnet"))
+            .expect("wallet to be initialized");
+        wallet::init_wallet(Network::Testnet, &temp.join("testnet"))
+            .expect("wallet to be initialized");
+        wallet::init_wallet(Network::Regtest, &temp.join("regtest"))
```
Comment (luckysori) [inline rust/src/wallet.rs:126]: ❓ So these tests were broken?

**Review by luckysori — APPROVED** [review]

File: rust/src/wallet.rs (hunk 2 above)
Comment (klochowicz) [reply rust/src/wallet.rs:126]: yup. got broken yesterday, seems like it :)

Comment (klochowicz) [reply rust/src/wallet.rs]: lol, they're still broken :D fixing up!

**Comment by klochowicz** [comment]
cc: @luckysori @holzeis I had to create a subdirectory per network type for the wallet to pass.

**Pull request merged**